src/main/java/io/cdap/plugin/format/MultiTableDBInputFormat.java

```diff
@@ -42,18 +40,16 @@
 import java.sql.Statement;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Input format that reads from multiple tables in a database using JDBC. Similar to Hadoop's DBInputFormat.
  */
-public class  MultiTableDBInputFormat extends InputFormat<NullWritable, StructuredRecord> {
+public class MultiTableDBInputFormat<T extends DBWritable> extends DataDrivenDBInputFormat<T> {
```

Comment:
I don't understand how this is working anymore if the input format is returning a DBWritable instead of a StructuredRecord. The source still expects the inputformat to return StructuredRecord

Reply:
Reverted to extend of InputFormat, and copied getSplitter() method from DataDrivenDBInputFormat.